signer/src/stacks/events.rs

```diff
@@ -505,10 +682,103 @@ mod tests {
         // let res = transform_value(value, NetworkKind::Regtest).unwrap();
         match transform_value(value, NetworkKind::Regtest).unwrap() {
             RegistryEvent::WithdrawalReject(event) => {
+                let expected_bitmap = BitArray::<[u8; 16]>::new(bitmap.to_le_bytes());
                 assert_eq!(event.request_id, request_id as u64);
-                assert_eq!(event.signer_bitmap, BitArray::<[u8; 16]>::new(bitmap.to_le_bytes()));
+                assert_eq!(event.signer_bitmap, expected_bitmap);
             }
             e => panic!("Got the wrong event variant: {e:?}"),
         };
     }
+
+    // Just a random public key to make the test case definitions below a
+    // little tidier.
+    static PUBLIC_KEY: LazyLock<CompressedPublicKey> = LazyLock::new(|| {
+        CompressedPublicKey(secp256k1::SecretKey::new(&mut OsRng).public_key(SECP256K1))
+    });
+
+    // A "tweaked" public key that is used to make the test case
+    // definition below a little easier on the eyes.
+    static TWEAKED_PUBLIC_KEY: LazyLock<TweakedPublicKey> =
+        LazyLock::new(|| TweakedPublicKey::dangerous_assume_tweaked((*PUBLIC_KEY).into()));
+
+    // A helper function for creating "P2SH-P2WPKH" and "P2SH-P2WSH" script
+    // hashes.
+    fn new_p2sh_segwit<T: AsRef<bitcoin::script::PushBytes>>(data: T) -> ScriptHash {
+        ScriptBuf::builder()
+            .push_int(0)
+            .push_slice(data)
+            .into_script()
+            .script_hash()
+    }
+
+    fn to_tuple_data<const N: usize>(version: u8, hash: [u8; N]) -> RawTupleData {
```

Comment:
Since this is a perfect conversion, this could be nice as a `From`- implementation

```
impl<const N: usize> From<(u8, [u8; N])> for RawTupleData
```

Alternatively as a constructor on the `RawTupleData` struct.

```
impl RawTupleData {
    fn new<const N: usize>(version: u8, hash: [u8; N]) -> Self {}
    }
```

Reply:
👍🏽 today I like the constructor approach.